plugins/modules/ipavault.py

```diff
@@ -39,123 +40,135 @@
 options:
   name:
     description: The vault name
+    type: list
+    elements: str
     required: true
     aliases: ["cn"]
   description:
     description: The vault description
+    type: str
     required: false
-  public_key:
+  vault_public_key:
     description: Base64 encode public key.
     required: false
-    type: string
-    aliases: ["ipavaultpublickey", "vault_public_key"]
-  public_key_file:
+    type: str
+    aliases: ["ipavaultpublickey", "public_key", "new_public_key"]
```

Comment:
We need to update "[README-vault.md](https://github.com/freeipa/ansible-freeipa/blob/master/README-vault.md)" for new_public_key.